platforms/core-configuration/declarative-dsl-evaluator/src/main/kotlin/org/gradle/internal/declarativedsl/evaluator/main/ConventionStorage.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.declarativedsl.evaluator.main
+
+import org.gradle.internal.declarativedsl.evaluator.conventions.ConventionDefinitionRegistrar
+import org.gradle.internal.declarativedsl.evaluator.conventions.SoftwareTypeConventionRepository
+import org.gradle.internal.declarativedsl.evaluator.conventions.SoftwareTypeConventionResolutionResults
+
+
+internal
+class ConventionStorage : ConventionDefinitionRegistrar, SoftwareTypeConventionRepository {
+    private
+    val conventionsMap = mutableMapOf<String, SoftwareTypeConventionResolutionResults>()
+
+    override fun registerConventions(conventionsBySoftwareType: Map<String, SoftwareTypeConventionResolutionResults>) {
+        // TODO: what if invoked multiple times?
+        conventionsMap.putAll(conventionsBySoftwareType)
+    }
```

Comment:
❓ Shouldn't we guard against multiple invocations until we know what to do?

Reply:
I was reluctant to do this because:

1. the client might reuse the evaluator instance when running analysis after edits in a settings updating the conventions. That, however, needs some special handling anyway, like, we should also be able to remove conventions from the map if they are no longer in the settings file. 
1. the client might want to register conventions from multiple sources – this is not something that we support now, but we may need this soon.

Anyway, both these cases need some more special handling.
I'd say for now a good solution might be to `conventionsMap.clear()` before this line, thus supporting (1) and not supporting (2).

Reply:
Added `conventionsMap.clear()` in https://github.com/gradle/gradle/pull/29276/commits/22261cebc5680850008efe97fcf32b45b2c83279.